samples/tacos/src/main/kotlin/com/slack/circuit/tacos/step/ConfirmationOrderStep.kt

```diff
@@ -0,0 +1,162 @@
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.withStyle
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.slack.circuit.tacos.OrderDetails
+import com.slack.circuit.tacos.R
+import com.slack.circuit.tacos.model.Diet
+import com.slack.circuit.tacos.model.Ingredient
+import java.math.BigDecimal
+import kotlinx.collections.immutable.ImmutableList
+import kotlinx.collections.immutable.toImmutableList
+
+object ConfirmationOrderStep : OrderStep {
+  override val index = 2
+  override val headerResId = R.string.confirm_step_header
+
+  data class OrderState(
+    val calories: Int,
+    val baseCost: BigDecimal,
+    val ingredientsCost: BigDecimal,
+    val diet: Diet,
+    val filling: String,
+    val toppings: ImmutableList<String>,
+  ) : OrderStep.State
+}
+
+@Composable
+internal fun confirmationProducer(orderDetails: OrderDetails): ConfirmationOrderStep.OrderState {
+  val toppings = orderDetails.toppings.sortedBy { it.name }.toImmutableList()
+  return ConfirmationOrderStep.OrderState(
+    calories = calculateCalories(orderDetails.filling, orderDetails.toppings),
+    baseCost = orderDetails.baseCost,
+    ingredientsCost = orderDetails.ingredientsCost,
+    diet = determineDiet(orderDetails.filling, orderDetails.toppings),
+    filling = orderDetails.filling.name,
+    toppings = toppings.map { it.name }.toImmutableList(),
+  )
+}
+
+private fun calculateCalories(filling: Ingredient, toppings: Set<Ingredient>): Int {
+  val toppingsTotal = toppings.fold(0) { acc, topping -> acc + topping.calories }
+  return toppingsTotal + filling.calories
+}
+
+private fun determineDiet(filling: Ingredient, toppings: Set<Ingredient>): Diet {
+  val updater = { curDiet: Diet, ingredient: Ingredient ->
+    when (ingredient.diet.ordinal < curDiet.ordinal) {
+      true -> ingredient.diet
+      false -> curDiet
+    }
+  }
+
+  // Iterate over the list of toppings and determine the highest common dietary level
+  // VEGAN being the highest, NONE being the lowest
+  val diet = toppings.fold(Diet.values().last(), updater)
+
+  // Check if the filling impacts the dietary level...
+  return updater(diet, filling)
+}
+
+@Composable
+internal fun ConfirmationUi(
+  state: ConfirmationOrderStep.OrderState,
+  modifier: Modifier = Modifier
+) {
+  with(state) {
+    val summary = buildSummary(filling)
+    val extra = buildExtras(diet, calories)
+
+    Column(modifier = modifier.fillMaxSize().padding(5.dp)) {
```

Comment:
these odd-numbered padding values are strange to me. Why not increments of 4/8 like normal?

Reply:
No reason. Updated to use increments of 4. https://github.com/slackhq/circuit/pull/496/commits/eee5a45f5db6e43f89110642e8c1955cb202333f